crates/hir-ty/src/diagnostics/decl_check.rs

```diff
@@ -551,6 +555,75 @@ impl<'a> DeclValidator<'a> {
         }
     }
 
+    /// Check incorrect names for fields of enum variant.
+    fn validate_enum_variant_fields(&mut self, variant_id: EnumVariantId) {
```

Comment:
This is copied from `validate_struct_fields`, and similar looking code now exists in a few places in this file. As such, I think it would be useful to clean this up in a follow-up PR, extracting the shared logic.